### DISCUSSION TITLE
Upgrade from 1.2.0-rc2

### DISCUSSION THREAD

**Discussion by kouinkouin** [modoboa/modoboa#2482] [Help]
Hi,

I have an (very) old instance of Modoboa "Modoboa 1.2.0-rc2 is currently in use." .
What is the best way to upgrade to the last release?

I guess create a machine from scratch is the fastest solution, but I don't know if there is a better solution.

Thanks!

**Comment by tonioo** [comment]
Hi, sorry for the late reply.
Indeed, you're using a pretty old version of Modoboa! I guess creating a new machine is not something you can avoid, but you'll still have to migrate your database and you might have some difficulties there...
I suggest you try on a new server with a copy of your database, and make sure to follow each migration steps for every version between yours and latest one. (https://modoboa.readthedocs.io/en/latest/upgrade.html#specific-instructions)

**Reply by kouinkouin** [reply]
Hi, no problem for the late reply, thanks for the reply!

So, on the new server, have I to install the 1.2.0-rc2 then upgrade version by version, or install the 1.7 (or 2.0 !) and play upgrade steps version by version?

**Reply by tonioo** [reply]
Yes, that's what I would try to do

**Reply by kouinkouin** [reply]
I guess it's a yes for the second idea (_install the 1.7 (or 2.0 !) and play upgrade steps version by version_) :-) (i don't want begin migrations with python2 and switch to python3 on a fresh install :-D )

**Reply by tonioo** [reply]
Now that you mention it, you might have issue with old modoboa versions because they are not compatible with python3. Maybe you can try to install 2.0 directly and try to apply upgrade steps of each version sequentially

**Reply by kouinkouin** [reply]
ok, I will do that in the next days/weeks. Thanks for the feedback!
I let you know of my (successful?) migration!
